x-pack/plugins/security_solution/public/detections/components/rules/rule_preview/use_preview_route.tsx

```diff
@@ -22,6 +23,9 @@ interface PreviewRouteParams {
   ruleType: Type;
   timeFrame: Unit;
   threatMapping: ThreatMapping;
+  threshold: FieldValueThreshold;
+  machineLearningJobId: string[];
+  anomalyThreshold: number;
```

Comment:
Mixing params from different rule types is something of an anti-pattern IMO.

Reply:
From a UI perspective, these are always present since it's just the form state. They do get separated into their different params later. There's probably better ways to pass that through though